include/jemalloc/internal/jemalloc_internal_defs.h.in

```diff
@@ -201,6 +201,11 @@
 /* One page is 2^LG_PAGE bytes. */
 #undef LG_PAGE
 
+#ifdef JEMALLOC_OVERRIDE_LG_PAGE
+    #undef LG_PAGE
+    #define LG_PAGE JEMALLOC_OVERRIDE_LG_PAGE
+#endif
+
```

Comment:
This actually won't work, since the `.h.in` is a template file for autoconf to generate the header. The output looks this way:
```
#ifdef JEMALLOC_OVERRIDE_LG_PAGE
    #define LG_PAGE 12
    #define LG_PAGE 12
#endif
```

For details see https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.71/html_node/Header-Templates.html 

Unfortunately it doesn't look like think there's a good way to do this in the template header.

Reply:
I can work around this by `#include`'ing another file that contains just this `#ifdef ...`, as suggested in the documentation you linked. It's admittedly pretty ugly, but it would work. Is it still worth upstreaming this at that point? I'll defer to your judgement.

Reply:
I'm debating on this too. Any other defines we might want to override in this header? If so then I'd say another header could be justified.

Reply:
> Any other defines we might want to override in this header?

I can definitely see wanting to override `LG_QUANTUM` for similar reasons, but can't say that I've tried running any experiments with it as of yet.

Reply:
Discussed with @guangli-dai . It seems there's no great solution either way (which is kind of expected when 2 different build systems are mixed). The current one we are leaning towards is, adding another upstream header dedicated for overrides, to be included at the end of this template. Then that header can either be empty upstream, or include the `JEMALLOC_OVERRIDE_LG_PAGE` use case.

This would make our internal code sync less error prone, e.g. we won't lose the override changes when the headers are re-generated. WDYT @Svetlitski ?

Reply:
Even if it is a bit of a kludge, I agree that having the additional header upstream will make our internal sync less error prone, which is important. I will refactor this to use an additional header and push shortly.

Reply:
Alright, pushed a new version which has a separate header. I tested this internally to make sure it plays nice with Buck2, and everything works fine.